controllers/clusterversion_controller.go

```diff
@@ -604,6 +624,63 @@ func labelClientOperatorSubscription(c *ClusterVersionReconciler) error {
 		}
 	}
 
+	c.subscriptionChannel = sub.Spec.Channel
+
 	c.log.Info("successfully labelled ocs-client-operator subscription")
 	return nil
 }
+
+func (c *ClusterVersionReconciler) reconcileSubscription() error {
+
+	storageClients := &v1alpha1.StorageClientList{}
+	if err := c.list(storageClients); err != nil {
+		return fmt.Errorf("failed to list storageclients: %v", err)
+	}
+
+	var desiredChannel string
+	for idx := range storageClients.Items {
+		storageClient := &storageClients.Items[idx]
+		annotations := storageClient.GetAnnotations()
+		if annotatedChannel, ok := annotations[utils.DesiredSubscriptionChannelAnnotationKey]; ok {
+			if desiredChannel != "" && desiredChannel != annotatedChannel {
+				// not all storageclients desiring to be on same channel
+				return nil
+			}
+			desiredChannel = annotatedChannel
+		}
+	}
+
+	if desiredChannel == "" {
+		// no storageclients mention a desired channel, user can bump the channel manually if required
+		return nil
+	}
+
+	subscriptions := &opv1a1.SubscriptionList{}
+	// we are already setting this label during reconcile phases
```

Comment:
Not sure I understand this comment

Reply:
Again. I am not sure what the comment means

Reply:
The label that I'm using here to list the ocs-client-operator subscription is set in another reconciler method and I'm simply stating that in the comment.

Reply:
done.